src/user-management/interfaces/authenticate-with-session-cookie.interface.ts

```diff
@@ -0,0 +1,33 @@
+import { User } from './user.interface';
+
+export interface AccessToken {
+  sid: string;
+  org_id?: string;
+  role?: string;
+  permissions?: string[];
+}
+
+export type SessionCookieData = {
+  user: User;
+  accessToken: string;
+  refreshToken: string;
+  impersonator: string;
+};
+
+export type AuthenticateWithSessionCookieFailureReason =
+  | 'invalid_jwt'
+  | 'invalid_session_cookie'
+  | 'no_session_cookie_provided';
```

Comment:
@PaulAsjes Since we'll be a bit locked in for awhile once we add, any edits to these reasons?

Reply:
Good idea, adding.

Reply:
Decided for all OAuth errors it may be more complete for us to pass the API error code through as the reason.

Reply:
I wonder if these should be enums that we export. In the rbac example project we do:

```
// If no session, redirect the user to the login page
if (!authenticated && reason === "no_session_cookie_provided") {
  return res.redirect("/login");
}
```

Better DX/less error prone would be something like 
```
import { AuthenticationmErrors } from '@workos-inc/node';

// If no session, redirect the user to the login page
if (!authenticated && reason === AuthenticationErrors.NO_SESSION_COOKIE_PROVIDED) {
  return res.redirect("/login");
}
```

Reply:
Good question. There isn't a huge difference between string literals and enums. Both will give the same auto-completion benefits and can be used for exhaustive checks. Enums of course have to be imported whereas string literals are provided via IntelliSense automatically.

Reply:
Updated these as enums! Kept it specific to `authenticateWithSessionCookie` since it's not relevant for other authentication methods.